src/posix/platform/spi_interface.cpp

```diff
@@ -606,16 +597,12 @@ otError SpiInterface::PushPullSpi(void)
     return error;
 }
 
-bool SpiInterface::CheckInterrupt(void)
-{
-    return (mIntGpioValueFd >= 0) ? (GetGpioValue(mIntGpioValueFd) == kGpioIntAssertState) : true;
-}
+bool SpiInterface::CheckInterrupt(void) { return (GetGpioValue(mIntGpioValueFd) == kGpioIntAssertState); }
 
 void SpiInterface::UpdateFdSet(void *aMainloopContext)
 {
-    struct timeval        timeout        = {kSecPerDay, 0};
-    struct timeval        pollingTimeout = {0, kSpiPollPeriodUs};
```

Comment:
We can also remove the constant `kSpiPollPeriodUs` fully (in `spi_interface.hpp`) since it wont be used anymore?